File: backend/Models/bookModel.js
const db = require('../Database/database');

const BookModel = {
    getAllBooks: (callback) => {
        db.all('SELECT * FROM books', [], callback);
    },

    getBookById: (id, callback) => {
        db.get('SELECT * FROM books WHERE id = ?', [id], callback);
    },

    createBook: (book, callback) => {
        const query = 'INSERT INTO books (title, author, year, image) VALUES (?, ?, ?, ?)';
        db.run(query, [book.title, book.author, book.year, book.image], function (err) {
            callback(err, this);
        });
    },

    updateBook: (id, book, callback) => {
        const query = 'UPDATE books SET title = ?, author = ?, year = ?, image = ? WHERE id = ?';
        db.run(query, [book.title, book.author, book.year, book.image, id], callback);
    },

    deleteBook: (id, callback) => {
        db.run('DELETE FROM books WHERE id = ?', [id], callback);
    },
};

module.exports = BookModel;

File: frontend_flutter/lib/Models/book_model.dart
class Book {
  final int id;
  final String title;
  final String author;
  final int year;
  final String? image;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.year,
    this.image,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'],
      title: json['title'],
      author: json['author'],
      year: json['year'],
      image: json['image'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'author': author,
      'year': year,
      'image': image,
    };
  }
}